managers/backup-manager/BackupManager.js

```diff
@@ -0,0 +1,489 @@
+
+    function isFinished(state) {
+      return _.includes(['succeeded', 'failed', 'aborted'], state);
+    }
+
+    return this.agent
+      .getBackupLastOperation(agent_ip)
+      .tap(lastOperation => {
+        if (isFinished(lastOperation.state)) {
+          return this.agent
+            .getBackupLogs(agent_ip)
+            .tap(logs => _.each(logs, log => logger.info(`Backup log for: ${opts.instance_guid} - ${JSON.stringify(log)}`)))
+            .then(logs => this.backupStore
+              .patchBackupFile(options, {
+                state: lastOperation.state,
+                logs: logs,
+                snapshotId: lastOperation.snapshotId
+              })
+            );
+        }
+      });
+  }
+
+  getLastBackup(tenant_id, instance_guid, noCache) {
+    return this.backupStore
+      .getBackupFile({
+        tenant_id: tenant_id,
+        service_id: this.service.id,
+        // plan_id: this.plan.id,
+        instance_guid: instance_guid
+      })
+      .then(metadata => {
+        switch (metadata.state) {
+        case 'processing':
+          return noCache ? this.agent
+            .getBackupLastOperation(metadata.agent_ip)
+            .then(data => _.assign(metadata, _.pick(data, 'state', 'stage'))) : metadata;
+        default:
+          return metadata;
+        }
+      });
+  }
+
+  abortLastBackup(tenant_id, instance_guid, force) {
+    logger.info('Aborting last backup', tenant_id, instance_guid);
+    return this.backupStore
+      .getBackupFile({
+        tenant_id: tenant_id,
+        service_id: this.service.id,
+        // plan_id: this.plan.id,
+        instance_guid: instance_guid
+      })
+      .tap(metadata => {
+        return eventmesh.server.updateAnnotationState({
+          resourceId: instance_guid,
+          annotationName: 'backup',
+          annotationType: 'default',
+          annotationId: metadata.backup_guid,
+          stateValue: 'aborting'
+        })
+      })
+      .then(metadata => {
+        if (!force && metadata.trigger === CONST.BACKUP.TRIGGER.SCHEDULED) {
+          throw new Forbidden('System scheduled backup runs cannot be aborted');
+        }
+        logger.info('metadata.state', metadata.state);
+        switch (metadata.state) {
+        case 'processing':
+          return this.agent
+            .abortBackup(metadata.agent_ip)
+            .return({
+              state: 'aborting'
+            });
+        default:
+          return _.pick(metadata, 'state');
+        }
+      });
+  }
+
+  static get prefix() {
+    return _
+      .reduce(config.directors,
+        (prefix, director) => director.primary === true ? director.prefix : prefix,
+        null) || super.prefix;
+  }
+
+  static parseDeploymentName(deploymentName, subnet) {
+    return _
+      .chain(utils.deploymentNameRegExp(subnet).exec(deploymentName))
+      .slice(1)
+      .tap(parts => parts[1] = parts.length ? parseInt(parts[1]) : undefined)
+      .value();
+  }
+}
+
+class Fabrik {
```

Comment:
We already have a Fabrik class. Creating it again here is confusing. We can directly expose this BackupManager class and access it.